release/release.go

```diff
@@ -81,51 +128,107 @@ func GenReleaseNotes(ctx context.Context, repo, milestone, prevMilestone string,
 	markdownVersion := strings.Replace(k8sVersion, ".", "", -1)
 	tmp := strings.Split(strings.Replace(k8sVersion, "v", "", -1), ".")
 	majorMinor := tmp[0] + "." + tmp[1]
-	fullVersion := majorMinor + "." + tmp[2]
 	changeLogSince := strings.Replace(strings.Split(prevMilestone, "+")[0], ".", "", -1)
 	sqliteVersionK3S := goModLibVersion("go-sqlite3", repo, milestone)
 	sqliteVersionBinding := sqliteVersionBinding(sqliteVersionK3S)
+	helmControllerVersion := goModLibVersion("helm-controller", repo, milestone)
+	coreDNSVersion := imageTagVersion("coredns", repo, milestone)
+	cgData := changeLogData{
+		PrevMilestone: prevMilestone,
+		Content:       content,
+	}
+
+	if repo == k3sRepo {
+		return genK3SReleaseNotes(
+			tmpl,
+			milestone,
+			k3sReleaseNoteData{
+				Milestone:             milestoneNoRC,
+				MajorMinor:            majorMinor,
+				K8sVersion:            k8sVersion,
+				ChangeLogVersion:      markdownVersion,
+				ChangeLogSince:        changeLogSince,
+				SQLiteVersion:         sqliteVersionBinding,
+				SQLiteVersionReplaced: strings.ReplaceAll(sqliteVersionBinding, ".", "_"),
+				HelmControllerVersion: helmControllerVersion,
+				ChangeLogData:         cgData,
+				CoreDNSVersion:        coreDNSVersion,
+			},
+		)
+	} else if repo == rke2Repo {
+		return genRKE2ReleaseNotes(
+			tmpl,
+			milestone,
+			rke2ReleaseNoteData{
+				MajorMinor:            majorMinor,
+				Milestone:             milestoneNoRC,
+				ChangeLogVersion:      markdownVersion,
+				K8sVersion:            k8sVersion,
+				HelmControllerVersion: helmControllerVersion,
+				CoreDNSVersion:        coreDNSVersion,
+				ChangeLogData:         cgData,
+			},
+		)
+	}
+	return nil, errors.New("invalid repo, it must be either k3s or rke2")
+}
+
+func genK3SReleaseNotes(tmpl *template.Template, milestone string, rd k3sReleaseNoteData) (*bytes.Buffer, error) {
+	tmpl = template.Must(tmpl.Parse(k3sReleaseNoteTemplate))
+	var runcVersion string
+	var containerdVersion string
+
+	if semver.Compare(rd.K8sVersion, "v1.24.0") == 1 &&
+		semver.Compare(rd.K8sVersion, "v1.26.5") == -1 {
```

Comment:
This can be on the same line as above. The creator of Go made clear that the language doesn't care about line length.